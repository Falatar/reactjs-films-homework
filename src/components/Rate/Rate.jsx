import React from 'react';
import PropTypes from 'prop-types';
import style from './Rate.scss';

function Rate({
  rating,
}) {
  return (
    <div className={style.rate}>
      <span className={rating > 0.5 ? style.star : style.voidStar} />
      <span className={rating > 1.5 ? style.star : style.voidStar} />
      <span className={rating > 2.5 ? style.star : style.voidStar} />
      <span className={rating > 3.5 ? style.star : style.voidStar} />
      <span className={rating > 4.5 ? style.star : style.voidStar} />
    </div>
  );
}

Rate.defaultProps = {
  rating: 0.0,
};

Rate.propTypes = {
  rating: PropTypes.number,
};

export default Rate;
