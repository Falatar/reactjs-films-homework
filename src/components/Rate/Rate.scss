div.rate {
    display: flex;
    flex-direction: row;
    border: 0;
    padding: 0;
    margin: 0;
}

.rate {
    span.star:before {
        content: '★';
    }

    span.voidStar:before {
        content: '☆';
    }

    span {
        display: flex;
        flex-direction: column;
        align-items: center;
        border: 0;
        padding: 0;
        margin: 0;
        color: deepskyblue;
        font-size: 1.7em;
        text-align: center;
        height: 1.7em;
    }
}