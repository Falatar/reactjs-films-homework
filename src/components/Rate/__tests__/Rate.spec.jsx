import React from 'react';
import renderer from 'react-test-renderer';
import Rate from '../Rate';

describe('Rate', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Rate rating="3.5" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
