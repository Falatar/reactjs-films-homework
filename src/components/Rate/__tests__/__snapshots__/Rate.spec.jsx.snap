// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="rate"
>
  <span
    className="voidStar"
  />
  <span
    className="voidStar"
  />
  <span
    className="voidStar"
  />
  <span
    className="voidStar"
  />
  <span
    className="voidStar"
  />
</div>
`;
